apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -482,7 +483,15 @@ export class ProjectBuildCache {
       hash.update(RushConstants.hashDelimiter);
     }
 
-    const projectStateHash: string = hash.digest('hex');
+    // The rebuild command is a special-case, and can use a cache from the build command.
+    const cacheableCommandName: string = options.commandName === 'rebuild' ? 'build' : options.commandName;
```

Comment:
The commandName should be the name of the underlying npm script, not the name passed to Rush. Then we won't need to special case this, and it will work with the multi-phase build proposal.

Reply:
@dmichon-msft when you say should, are you suggesting a change be made in this pull request? Not sure what the suggestion is here exactly.